Duplicati/CommandLine/Commands.cs

```diff
@@ -115,7 +115,7 @@ public void Dispose()
 
                         if (m_thread != null && m_thread.IsAlive)
                         {
-                            m_thread.Abort();
+                            m_thread.Interrupt();

```

Comment:
Thread.Interrupt() on ends a thread when it hits a thread wait like sleep() or join().  It also throws different excepctions than Abort().  So exits will be a lot slower, or possibly wait until the thread is complete.

You indicate the exceptions aren't handled, so maybe that's not a problem.  I've not reviewed all the relevant code to be sure.

However the interrupt on Stopnow interruption test intermittently failing does look related to threads management of ending.  Only some of them have a cancellation token, so I feel that is related but I have not hard evidence for that.

Is cancellation token the desired long term solution with this and an intermediate step?

Reply:
Cancellation token would be ideal.

Reply:
As a test, I removed Thread.Abort code completely and have been running the StopNow() test.  It fails a lot less when I did that.  But still fails sometimes, I'm still investigating and you can foolow that in the forum thread.